firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/testutil/IntegrationTestUtil.kt

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.testutil
+
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.FirebaseFirestoreSettings
+import com.google.firebase.firestore.ktx.BuildConfig
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.TimeUnit
+
+/**
+ * Whether the integration tests should run against a local Firestore emulator or the Production
+ * environment
+ */
+private const val CONNECT_TO_EMULATOR = BuildConfig.USE_EMULATOR_FOR_TESTS
+
+/** Default amount of time to wait for a given operation to complete, used by waitFor() helper. */
+private const val OPERATION_WAIT_TIMEOUT_MS = 30000L
+
+private const val AUTO_ID_LENGTH = 20
+
+private val AUTO_ID_ALPHABET = ('A'..'Z') + ('a'..'z') + ('0'..'9')
+
+private var settingWithEmulator =
```

Comment:
Use https://github.com/firebase/firebase-android-sdk/blob/master/firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt#L151 for creating setting objects

Reply:
Thank you for the reference, I am using the Kotlin extension function creating Emulator settings now. 
I also found `.setPersistenceEnabled(true)` is default, so I can skip this line to make the code shorter.